sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubConsumerClientConfiguration.java

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.eventhubs;
+
+import com.azure.messaging.eventhubs.EventHubClientBuilder;
+import com.azure.messaging.eventhubs.EventHubConsumerAsyncClient;
+import com.azure.messaging.eventhubs.EventHubConsumerClient;
+import com.azure.spring.cloud.autoconfigure.eventhubs.factory.EventHubClientBuilderFactory;
+import com.azure.spring.core.ApplicationId;
+import com.azure.spring.core.StaticConnectionStringProvider;
+import com.azure.spring.core.service.AzureServiceType;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+
+import static com.azure.spring.cloud.autoconfigure.context.AzureContextUtils.EVENT_HUB_CONSUMER_CLIENT_BUILDER_BEAN_NAME;
+import static com.azure.spring.cloud.autoconfigure.context.AzureContextUtils.EVENT_HUB_CONSUMER_CLIENT_BUILDER_FACTORY_BEAN_NAME;
+
+/**
+ * Configuration for Event Hub consumer client, which provides {@link EventHubConsumerClient} or
+ * {@link EventHubConsumerAsyncClient}.
+ *
+ */
+@Configuration(proxyBeanMethods = false)
+@Import({
+    AzureEventHubConsumerClientConfiguration.DedicatedConsumerConnectionConfiguration.class,
+    AzureEventHubConsumerClientConfiguration.SharedConsumerConnectionConfiguration.class
+})
+@ConditionalOnExpression(
```

Comment:
what does this condition is actually doing? which codition?

Reply:
at least one eventhub-name is provided for us to configure the consumer.

Reply:
As least one eventhub name is provided.